joomla/joomla-cms
# Remove overrides when a template is uninstalled

Pull Request for #38700 

## The Problem

When you uninstall a template it does NOT remove the database entry for any overrides for that template in #__template_overrides.

As a result the count on the quickicon does not change.

This site was a j3 site that has now been updated to j4

## Before
![image](https://user-images.githubusercontent.com/1296369/188847528-8774b293-5ed4-4568-914e-4aa6befcbe66.png)
![image](https://user-images.githubusercontent.com/1296369/188847561-784eb6db-00c4-4893-a2eb-c40f162455fe.png)

## After uninstalling the template (called editor)
![image](https://user-images.githubusercontent.com/1296369/188847667-7cd6f148-e7e4-4eec-a319-af8f4d1a3b5a.png)
![image](https://user-images.githubusercontent.com/1296369/188847679-1cb0acac-b49b-4c65-bb5a-981ce303f1bb.png)
![image](https://user-images.githubusercontent.com/1296369/188847829-5c075d6d-334e-4b9c-aa84-eb2382110a64.png)

## The solution
### Part 1
When a template is uninstalled any record in the #__template_overrides table for that template will now be  deleted

### Part 2
On update the #__template_overrides is checked to see if there are any orphaned records from previous templates that have now been deleted and if found then it deletes those records.

## Test instructions
Create a site with a previous release of joomla
Clone the template TWICE
In each cloned template create various template overrides
Update the site to the current release of joomla
You should now see a count for the number of overrides to check on the home dashboard
Uninstall one of the cloned templates
You should now see that the count for the number of overrides to check on the home dashboard has not changed
Update the site to this build using this update url https://ci.joomla.org/artifacts/joomla/joomla-cms/4.2-dev/38714/downloads/57633/pr_list.xml
You should now see that the count for the number of overrides to check on the home dashboard has changed
- that is a successful test of part 2
Finally uninstall the remaining clone template
You should now see that the count for the number of overrides to check on the home dashboard has changed to none
- that is a successful test of part 1
